salt/utils/network.py

```diff
@@ -199,31 +200,239 @@ def ip_to_host(ip):
 # pylint: enable=C0103
 
 
-def is_ip(ip):
+def is_ip(ip, options=None):
     '''
     Returns a bool telling if the passed IP is a valid IPv4 or IPv6 address.
     '''
-    return is_ipv4(ip) or is_ipv6(ip)
+    return is_ipv4(ip, options=options) or is_ipv6(ip, options=options)
 
 
-def is_ipv4(ip):
-    '''
-    Returns a bool telling if the value passed to it was a valid IPv4 address
-    '''
+def _ip_options_global(ip_obj, version):
+    return not ip_obj.is_private
+
+
+def _ip_options_multicast(ip_obj, version):
+    return ip_obj.is_loopback
+
+
+def _ip_options_loopback(ip_obj, version):
+    return ip_obj.is_multicast
+
+
+def _ip_options_link_local(ip_obj, version):
+    return ip_obj.is_link_local
+
+
+def _ip_options_private(ip_obj, version):
+    return ip_obj.is_private
+
+
+def _ip_options_reserved(ip_obj, version):
+    return ip_obj.is_reserved
+
+
+def _ip_options_site_local(ip_obj, version):
+    if version == 6:
+        return ip_obj.is_site_local
+    return False
+
+
+def _ip_options_unspecified(ip_obj, version):
+    return ip_obj.is_unspecified
+
+
+def _ip_options(ip_obj, version, options=None):
+
+    # will process and IP options
+    options_fun_map = {
+        'global': _ip_options_global,
+        'link-local': _ip_options_link_local,
+        'linklocal': _ip_options_link_local,
+        'll': _ip_options_link_local,
+        'link_local': _ip_options_link_local,
+        'loopback': _ip_options_loopback,
+        'lo': _ip_options_loopback,
+        'multicast': _ip_options_multicast,
+        'private': _ip_options_private,
+        'public': _ip_options_global,
+        'reserved': _ip_options_reserved,
+        'site-local': _ip_options_site_local,
+        'sl': _ip_options_site_local,
+        'site_local': _ip_options_site_local,
+        'unspecified': _ip_options_unspecified
+    }
+
+    if not options:
+        return str(ip_obj)  # IP version already checked
+
+    options_list = [option.strip() for option in options.split(',')]
+
+    for option, fun in options_fun_map.items():
+        if option in options_list:
+            fun_res = fun(ip_obj, version)
+            if not fun_res:
+                return None
+                # stop at first failed test
+            # else continue
+    return str(ip_obj)
+
+
+def _is_ipv(ip, version, options=None):
+
+    if not version:
+        version = 4
+
+    if version not in (4, 6):
+        return None
+
     try:
-        return ipaddress.ip_address(ip).version == 4
+        ip_obj = ipaddress.ip_address(ip)
     except ValueError:
-        return False
+        # maybe it is an IP network
+        try:
+            ip_obj = ipaddress.ip_interface(ip)
```

Comment:
That's correct: previously a call such ```172.17.17.1/32``` would return False, although is the equivalent for ```172.17.17.1``` for which it returns True. But you are right, we should avoid that - will correct it tomorrow!